### ISSUE TITLE
Add all options to configuration object

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#5142]
- [x] [`report*`](https://github.com/stylelint/stylelint/issues/3858)
- [ ] [`files`](https://github.com/stylelint/stylelint/issues/3860)
- [ ] [`fix`](https://github.com/stylelint/stylelint/issues/3859)
- [ ] [`customSyntax`](https://github.com/stylelint/stylelint/issues/4843)

There maybe more, but we can start there.

Once these are in place we can approach the `overrides` property (https://github.com/stylelint/stylelint/issues/3128).

And once that's in place, we can decide what approach we're going to take with non-standard syntax, i.e. continue with trying to autodetect, like Prettier, or move to configuration-based approach, like ESLint.

**Comment by ybiquitous** [comment]
@jeddy3 What status is the checklist in this issue description?

**Comment by jeddy3** [comment]
I've:
- checked `customSyntax` as done
- removed `maxWarnings` as the original request was closed by the author. 

I believe the list is accurate now. 

---

Now that `14.0.0` has been released, I'm going to make some time over the coming weeks to help go through our open issues as I suspect a number of them should either be closed or updated.

**Comment by Issayev** [comment]
Is it possible to add  config file 'ignorePath' option support in the future?  It has been already asked earlier in #4126 along with 'ignorePattern' option. I don't know current state of the second option, but 'ignorePath'  doesn't work when it's being included in config file.

**Comment by oliviertassinari** [comment]
+1 for `ignorePath`. I would like to be able to do

```jsx
export default {
  extends: 'stylelint-config-standard',
  ignorePath: ['.gitignore', '.eslintignore'],
``` 

For two reasons:

1. So that https://github.com/stylelint/vscode-stylelint is aware to not lint the file when it opens it (right now it reports confusing errors)
2. So that I can share this config with other repositories, without having to duplicate --ignore-path.

**Comment by jeddy3** [comment]
@oliviertassinari Thanks for chiming in with your use case. I've [created an issue](https://github.com/stylelint/stylelint/issues/8345) for us all to discuss.

**Comment by Mouvedia** [comment]
In light of #8422, I have untick `cache`.
